cdap-security/src/main/java/io/cdap/cdap/security/authorization/AccessControllerWrapper.java

```diff
@@ -0,0 +1,267 @@
+    this.accessControllerV1 = accessControllerV1;
+  }
+
+  @Override
+  public void initialize(AuthorizationContext context) {
+    try {
+      accessControllerV1.initialize(context);
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
+    }
+  }
+
+  @Override
+  public AuthorizationResponse createRole(Principal caller, Role role) throws AccessException {
+    try {
+      accessControllerV1.createRole(role);
+    } catch (UnauthorizedException e) {
+        return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizationResponse dropRole(Principal caller, Role role) throws AccessException {
+    try {
+      accessControllerV1.dropRole(role);
+    } catch (UnauthorizedException e) {
+      return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizationResponse addRoleToPrincipal(Principal caller, Role role, Principal principal)
+    throws AccessException {
+    try {
+      accessControllerV1.addRoleToPrincipal(role, principal);
+    } catch (UnauthorizedException e) {
+      return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizationResponse removeRoleFromPrincipal(Principal caller, Role role, Principal principal)
+    throws AccessException {
+    try {
+      accessControllerV1.removeRoleFromPrincipal(role, principal);
+    } catch (UnauthorizedException e) {
+      return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizedResult<Set<Role>> listRoles(Principal caller, Principal principal) throws AccessException {
+    try {
+      Set<Role> roleSet = accessControllerV1.listRoles(principal);
+      return new AuthorizedResult<>(roleSet, AuthorizationResponse.Builder.defaultNotRequired());
+    } catch (UnauthorizedException e) {
+      //Incase of UnauthorizedException , the set will be empty
+      return new AuthorizedResult<>(Collections.emptySet(), createAuthResultFromUnauthorizedExp(e));
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public AuthorizedResult<Set<Role>> listAllRoles(Principal caller) throws AccessException {
+    try {
+      Set<Role> roleSet = accessControllerV1.listAllRoles();
+      return new AuthorizedResult<>(roleSet, AuthorizationResponse.Builder.defaultNotRequired());
```

Comment:
This would make us count this as "BYPASS". It's better to assume that extension did authorize the operation